src/parse-mockdb.js

```diff
@@ -104,9 +104,24 @@ function runHook(className, hookType, data) {
     const modelData = Object.assign(new Object, data, {className});
     const model = Parse.Object.fromJSON(modelData);
 
-    return hook.bind(model)().then((result) => {
-      debugPrint('HOOK', result);
-      return Parse.Promise.as(result.toJSON());
+    // TODO Stub out Parse.Cloud.useMasterKey() so that we can report the correct 'master'
+    // value here.
+    var beforeSaveOrBeforeDeleteRequestObject = {
+      installationId: 'parse-mockdb',
+      master: false,
+      object: model,
+      user: "ParseMockDB doesn't define request.user."
+    };
+
+    return hook(beforeSaveOrBeforeDeleteRequestObject).done((beforeSaveOverrideValue) => {
+      debugPrint('HOOK', { beforeSaveOverrideValue });
+
+      // Unlike BeforeDeleteResponse, BeforeSaveResponse might specify
+      var objectToProceedWith =  hookType === 'beforeSave' && beforeSaveOverrideValue
```

Comment:
yes, i don't see any reason to make a copy.  The model is passed by reference and it is a copy, so if there's a reason, I sure can't grok it.  I played around with the result using the unit test and i can't think of a test case to write where it would make a difference.